modules/dnn/src/dnn.cpp

```diff
@@ -3205,15 +3227,51 @@ Net Net::Impl::createNetworkFromModelOptimizer(InferenceEngine::CNNNetwork& ieNe
 #ifdef HAVE_DNN_NGRAPH
         if (DNN_BACKEND_INFERENCE_ENGINE_NGRAPH == getInferenceEngineBackendTypeParam())
         {
+            const auto& outputName = it.first;
             Ptr<Layer> cvLayer(new NgraphBackendLayer(ieNet));
+            cvLayer->name = outputName;
+            cvLayer->type = "_unknown_";
 
-            InferenceEngine::CNNLayerPtr ieLayer = ieNet.getLayerByName(it.first.c_str());
-            CV_Assert(ieLayer);
+            if (ngraphFunction)
+            {
+                CV_TRACE_REGION("ngraph_function");
+                bool found = false;
+                for (const auto op : ngraphOperations)
```

Comment:
Can we remove this variable? Just
```cpp
for (const auto& op : ngraphFunction->get_ops())
```

Reply:
`get_ops()` is not trivial. So it is better to cache its result while using in a loop body.

Reply:
Can we cache this right before loop?
```
auto ngraphOperations = ngraphFunction->get_ops();
for (const auto op : ngraphOperations)
```

Reply:
Probably, we have to change it here as well: https://github.com/opencv/opencv/blob/bfcc136dc7bd2e246a67be0a39e2922aacc85b4c/modules/dnn/src/ie_ngraph.cpp#L596?

Reply:
```
auto ngraphOperations = ngraphFunction->get_ops();
for (const auto op : ngraphOperations)
```

is the same as

```
for (const auto op : ngraphFunction->get_ops())
```

---

Currently it is cached before this parent loop:
```
for (auto& it : ieNet.getOutputsInfo())
```

---

> we have to change it here as well:

There is no parent loop in that case. Caching can't help in that case.

Reply:
Now I see it, thanks!

Reply:
One more question here is shall we use
```
for (const auto& op : ngraphOperations)
```
instead of
```cpp
for (const auto op : ngraphOperations)
```
?

Reply:
Replaced to `const auto&`.